src/_nebari/stages/kubernetes_services/template/modules/kubernetes/services/conda-store/config/conda_store_config.py

```diff
@@ -83,11 +88,301 @@ def conda_store_config(path="/var/lib/conda-store/config.json"):
 c.GenericOAuthAuthentication.user_data_key = "preferred_username"
 c.GenericOAuthAuthentication.tls_verify = False
 
+CONDA_STORE_ROLE_PERMISSIONS_ORDER = ["viewer", "developer", "admin"]
+
+
+@dataclasses.dataclass
+class CondaStoreNamespaceRole:
+    namespace: str
+    role: str
+
+
+@dataclasses.dataclass
+class KeycloakCondaStoreRoleScopes:
+    scopes: str
+    log: logging.Logger
+
+    def _validate_role(self, role):
+        valid = role in CONDA_STORE_ROLE_PERMISSIONS_ORDER
+        self.log.info(f"role: {role} is valid={valid}")
+        return valid
+
+    def parse_role_and_namespace(
+        self, text
+    ) -> typing.Optional[CondaStoreNamespaceRole]:
+        # The regex pattern
+        pattern = r"^(\w+)!namespace=([^!]+)$"
+
+        # Perform the regex search
+        match = re.search(pattern, text)
+
+        # Extract the permission and namespace if there is a match
+        if match and self._validate_role(match.group(1)):
+            return CondaStoreNamespaceRole(
+                namespace=match.group(2), role=match.group(1)
+            )
+        else:
+            return None
+
+    def parse_scope(self) -> typing.List[CondaStoreNamespaceRole]:
+        """Parsed scopes from keycloak role's attribute and returns a list of role/namespace
+        if scopes' syntax is valid otherwise return []
+
+        Example:
+            Given scopes as "viewer!namespace=scipy,admin!namespace=pycon", the function will
+            return [{"role": "viewer", "namespace": "scipy"}, {"role": "admin", "namespace": "pycon"}]
+        """
+        if not self.scopes:
+            self.log.info(f"No scope found: {self.scopes}, skipping role")
+            return []
+        scope_list = self.scopes.split(",")
```

Comment:
Can namespaces have a "," in them.  Can we restrict that or if not, add some docs saying that namespaces should not have commas in them.  Maybe check for commas in your parse_role_and_namespace function.

Reply:
Namespaces can't have `,` in them.
See: https://github.com/conda-incubator/conda-store/blob/67a4860ac5d9cf33285b5641e5dbb3b9ce61b1c7/conda-store-server/conda_store_server/_internal/schema.py#L24